### PR TITLE
Pull more versions with release notes from github releases

### PR DESCRIPTION
**Pull request by dariusj18**
This PR collects and displays additional changelog information on the version show page.

Previously the database would store a single entry for each major release.  These entries were updated with the latest minor and patch information.  

We now store an entry in the database for every tag/release.  To relate these versions a few columns were added to the database to make querying and setting up relationships easier.

- Global `front` scope added to limit results to only front versions
  - Overriding when updating releases, getting all releases for version, getting the `latest_version` for show routes, and in `LaravelVersionFromPath` when getting the requested version. 
- Changed some route handling 
  - invalid versions return 404 instead of trying to redirect (`/10.0.3.2` => `/10.0.3`)
  - Pre semver majors without minors are redirected instead of returning 404 (`/3` => `/3.0`)

Screenshot 2023-04-24 at 3 36 12 PM

### CODE REVIEW

**Comment by dariusj18** [comment]
I didn't touch the API side, I was mainly doing this because I saw an opportunity to add some extra information.

**Comment by mattstauffer** [comment]
I love this! I've been wanting to re-work it to add a DB entry for every release, like we do with Checkmate (https://github.com/tighten/checkmate), but you just got ahead of it for me, parsed the description, *and* added the concept of `is_front`!

Thank you!

I'll need to work on this, or you can, so we can add some tests to make sure that non `is_front` versions don't show up in the wrong places, and the checks that show some things on `/5.1.1` type routes but not on `/5` routes still work correctly, but we really needed those tests anyway. Also will have to check and test the API to make sure this didn't break it--i assume it'll need some scoping down before this can be merged.

Feel free to make any updates you want based on this paragraph above; i'll also try to find some time to work through it and implement some of these changes if i can.

Again, thanks!

File: README.md [hunk 1]
```diff
@@ -118,12 +118,14 @@ Potential statuses:
     ```bash
     php artisan key:generate
     ```
+1. Create a [GitHub Personal Access Token](https://github.com/settings/tokens) and update the  `GITHUB_TOKEN` key in your `.env` file.
```
Comment (faxblaster) [inline README.md:121]: Not sure if it's worth noting here, but you have to use a "classic" PAT for GraphQL queries. I tested first with a fine-grained token, but looks like [the github cli hasn't been updated to support them yet](https://github.com/cli/cli/issues/6680). 🙃

**Review by faxblaster — CHANGES_REQUESTED** [review]
Added a quick note for merging, but this is so nice! I especially like the new `version->order` concept for speed, rather than ordering on multiple attributes. And the tests are especially nice and readable. 🚀

**Comment by driftingly** [comment]
Thanks @dariusj18 for getting this started!  Sorry it sat for so long!

**Pull request merged**